### DISCUSSION TITLE
Getting 404 trying to deploy and not sure what the root of the issue is...

### DISCUSSION THREAD

**Discussion by JChurgin** [community/community#58235] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Show & Tell

### Body

this is the repo: https://github.com/JChurgin/vite-project-protfolio

could it be my repo name? 
its a vite project but I am pretty sure I have to workflow working.

will add screen shots too.
![‏‏צילום מסך (118)](https://github.com/community/community/assets/114018853/e26e80ba-0f78-4e7b-a015-998307aac5f8)
![‏‏צילום מסך (119)](https://github.com/community/community/assets/114018853/6983c31e-8e9f-45da-a316-b2de7dbcc3a0)
![‏‏צילום מסך (121)](https://github.com/community/community/assets/114018853/93d8f711-0c24-4804-9048-11a5576e658c)
![‏‏צילום מסך (122)](https://github.com/community/community/assets/114018853/3cbff57d-7df4-4ad7-99b5-49b3fac65fef)
![‏‏צילום מסך (124)](https://github.com/community/community/assets/114018853/fa11fe2e-a7d1-413c-bb24-82e22c746669)

**Comment by JChurgin** [comment]
I  did some changes adding gp-pages globally, trying to push again

**Reply by JChurgin** [reply]
Update still getting 404

**Reply by JChurgin** [reply]
got the gp-pages branch working and moved on to a new error:

Uncaught TypeError: Failed to resolve module specifier "react-icons/Bs". Relative references must start with either "/", "./", or "../".

**Comment by airtower-luna** [comment]
The problem is that your [`index.html` contains absolute URLs](https://github.com/JChurgin/vite-project-protfolio/blob/4342f0df79c192623ee78d1a029574aa4479313a/index.html#L11) both for that script and the icon (for the icon the path is also incorrect if interpreted from the repository root, it seems to exist in `img/`).

Your deployment ends up in a subdirectory named like the repository, so if you want to use absolute URLs it needs to include that before the path within the repository. A *better* solution would be to use relative URLs, so it works no matter where your site is deployed.

See also: https://developer.mozilla.org/en-US/docs/Learn/Common_questions/Web_mechanics/What_is_a_URL#absolute_urls_vs._relative_urls

**Reply by JChurgin** [reply]
could you point out what absolut URL you are talking about? 

Do you mean my json homepage?   "homepage": "https://JChurgin.github.io/vite-project-protfolio"?

isnt the URL a link ?     `<link rel="icon" type="image/svg+xml" href="/vite.svg" />`
I prob dont need it even.

**Reply by airtower-luna** [reply]
Wait a minute, your actually deployed `index.html` is quite different from the one in your repository, and actually (currently) looks okay. You seem to have two competing deployment workflows, both a custom one and the default Jekyll build. Maybe something got mixed up there? :thinking:

**Reply by JChurgin** [reply]
well with vite project you need a custom workflow is there a way to not let it not do a jekyll workflow?

at least per : https://vitejs.dev/guide/static-deploy.html

latest repo : https://github.com/JChurgin/vite-project-protfolio

FYI: i see the background getting deployed

**Reply by airtower-luna** [reply]
You'll need to set the publishing source to "GitHub Actions" in the repository pages settings: https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#publishing-with-a-custom-github-actions-workflow

**Reply by JChurgin** [reply]
ok so the deployment seem to work but am running in to an error `Uncaught TypeError: Failed to resolve module specifier "react-icons/Bs". Relative references must start with either "/", "./", or "../".` 

do you know what it is? or how i can find out what the root that is causing it?

**Reply by airtower-luna** [reply]
That sounds like something is either referenced incorrectly or missing, but I'm no React expert, so I can't really help there.
